src/internal/m365/collection/drive/collections_tree.go

```diff
@@ -0,0 +1,325 @@
+
+	// --- pager aggregation
+
+	// du, newPrevPaths, err := c.PopulateDriveCollections(
+	// 	ctx,
+	// 	d,
+	// 	tree,
+	// 	cl.Local(),
+	// 	errs)
+	// if err != nil {
+	// 	return nil, false, clues.Stack(err)
+	// }
+
+	// numDriveItems := c.NumItems - numPrevItems
+	// numPrevItems = c.NumItems
+
+	// cl.Add(count.NewPrevPaths, int64(len(newPrevPaths)))
+
+	// --- prev path incorporation
+
+	// For both cases we don't need to do set difference on folder map if the
+	// delta token was valid because we should see all the changes.
+	// if !du.Reset {
+	// 	if len(excludedItemIDs) == 0 {
+	// 		continue
+	// 	}
+
+	// 	p, err := c.handler.CanonicalPath(odConsts.DriveFolderPrefixBuilder(driveID), c.tenantID)
+	// 	if err != nil {
+	// 		return nil, false, clues.WrapWC(ictx, err, "making exclude prefix")
+	// 	}
+
+	// 	ssmb.Add(p.String(), excludedItemIDs)
+
+	// 	continue
+	// }
+
+	// Set all folders in previous backup but not in the current one with state
+	// deleted. Need to compare by ID because it's possible to make new folders
+	// with the same path as deleted old folders. We shouldn't merge items or
+	// subtrees if that happens though.
+
+	// --- post-processing
+
+	// Attach an url cache to the drive if the number of discovered items is
+	// below the threshold. Attaching cache to larger drives can cause
+	// performance issues since cache delta queries start taking up majority of
+	// the hour the refreshed URLs are valid for.
+
+	// if numDriveItems < urlCacheDriveItemThreshold {
+	// 	logger.Ctx(ictx).Infow(
+	// 		"adding url cache for drive",
+	// 		"num_drive_items", numDriveItems)
+
+	// 	uc, err := newURLCache(
+	// 		driveID,
+	// 		prevDeltaLink,
+	// 		urlCacheRefreshInterval,
+	// 		c.handler,
+	// 		cl,
+	// 		errs)
+	// 	if err != nil {
+	// 		return nil, false, clues.Stack(err)
+	// 	}
+
+	// 	// Set the URL cache instance for all collections in this drive.
+	// 	for id := range c.CollectionMap[driveID] {
+	// 		c.CollectionMap[driveID][id].urlCache = uc
+	// 	}
+	// }
+
+	return nil, nil, pagers.DeltaUpdate{}, clues.New("not yet implemented")
+}
+
+// quality-of-life wrapper that transforms each tombstone in the map
+// into a backup collection that marks the backup as deleted.
+func (c *Collections) makeDriveTombstones(
+	ctx context.Context,
+	driveTombstones map[string]struct{},
+	errs *fault.Bus,
+) ([]data.BackupCollection, error) {
+	c.counter.Add(count.DriveTombstones, int64(len(driveTombstones)))
+
+	var (
+		colls = make([]data.BackupCollection, 0, len(driveTombstones))
+		el    = errs.Local()
+	)
+
+	// generate tombstones for drives that were removed.
+	for driveID := range driveTombstones {
+		if el.Failure() != nil {
+			break
+		}
+
+		prevDrivePath, err := c.handler.PathPrefix(c.tenantID, driveID)
+		if err != nil {
+			err = clues.WrapWC(ctx, err, "making drive tombstone for previous path").Label(count.BadPathPrefix)
+			el.AddRecoverable(ctx, err)
+
+			continue
+		}
+
+		coll, err := NewCollection(
```

Comment:
nit: how about using the `data.NewTombstoneCollection()` helper to keep things consistent? Unless maybe we still rely on being able to cast to a onedrive specific implementation of the collection? In which case could you add a TODO here about that

Reply:
I can look at that later.  This is purely copy-paste from the original code, which is also using `NewCollection`.  I'll add a TODO for it.